core/src/main/java/io/grpc/transport/AbstractClientStream.java

```diff
@@ -45,9 +47,12 @@
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
+import javax.annotation.concurrent.NotThreadSafe;
+
 /**
  * The abstract base class for {@link ClientStream} implementations.
  */
+@NotThreadSafe
```

Comment:
This is the very first usage of `NotThreadSafe`. We currently follow the standard practice of "something isn't thread-safe unless it says it is thread-safe." Are you thinking that we should start putting this on all classes, or this one is special for some reason?